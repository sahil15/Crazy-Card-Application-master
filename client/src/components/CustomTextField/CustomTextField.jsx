import React from "react";
import TextField from "@material-ui/core/TextField";
import { connect } from "react-redux";
import { addCustomer } from "../../store/actions/customer";

class CustomTextField extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false,
    };
  }
  render() {
    const onChange = (e, field) => {
      let hasError = e.target.value.length < field.minLength ? true : false;

      this.setState({ hasError });
      this.props.addCustomer({ [field.name]: e.target.value });
    };

    return (
      <TextField
        error={this.state.hasError}
        helperText={this.state.hasError ? this.props.field.helperText : ""}
        minLength={this.props.field.minLength}
        required={this.props.field.required}
        label={this.props.field.label}
        type={this.props.field.type}
        defaultValue={this.props.field.defaultValue}
        className="center"
        onChange={(e) => onChange(e, this.props.field)}
      ></TextField>
    );
  }
}

const mapStateToProps = (state) => ({
  state: state,
});
const mapDispatchToProps = (dispatch) => ({
  addCustomer: (data) => {
    dispatch(addCustomer(data));
  },
});

export {CustomTextField};

export default connect(mapStateToProps, mapDispatchToProps)(CustomTextField);
