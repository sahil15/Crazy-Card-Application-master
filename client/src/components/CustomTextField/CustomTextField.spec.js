import React from 'react';
import { shallow } from 'enzyme';
import {CustomTextField} from './CustomTextField.jsx';

describe("Tests for CustomTextField ", ()=> {
    const mockField = {
            name: "dob",
            label: "Birthday",
            type: "date",
            defaultValue: "2017-05-24",
        };
    it('Renders correctly', () => {
        const wrapper = shallow(<CustomTextField field={mockField} />);
        expect(wrapper).toMatchSnapshot();
      });
})

   
   

