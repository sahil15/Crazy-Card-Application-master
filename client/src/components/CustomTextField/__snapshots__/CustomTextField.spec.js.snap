// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CustomTextField  Renders correctly 1`] = `
<WithStyles(ForwardRef(TextField))
  className="center"
  defaultValue="2017-05-24"
  error={false}
  helperText=""
  label="Birthday"
  onChange={[Function]}
  type="date"
/>
`;
